### ISSUE TITLE
Create a JSON schema for validation

### ISSUE THREAD

**Issue by alexander-held** [hep-statistics-serialization-standard/hep-statistics-serialization-standard#3]
A JSON schema (https://json-schema.org/) to validate against would be useful. It can help catch formatting issues, unknown keys, or other behavior that might cause subsequent problems.

**Comment by jonas-eschle** [comment]
I agree, I think I can take care of this towards September of this year. Possibly using [marshmallow](https://marshmallow.readthedocs.io/en/stable/) that provides an easy way to compose a scheme

**Comment by cburgard** [comment]
That would be amazing @jonas-eschle, thanks so much!

**Comment by nsmith-** [comment]
I have some experience in using [pydantic](https://docs.pydantic.dev/latest/) to define a JSON schema for [correctionlib](https://github.com/cms-nanoAOD/correctionlib/blob/master/src/correctionlib/schemav2.py). Happy to provide input

**Comment by jonas-eschle** [comment]
> I have some experience in using [pydantic](https://docs.pydantic.dev/latest/) to define a JSON schema for [correctionlib](https://github.com/cms-nanoAOD/correctionlib/blob/master/src/correctionlib/schemav2.py?rgh-link-date=2023-07-20T13%3A30%3A54Z) (and [auto-docs](https://cms-nanoaod.github.io/correctionlib/schemav2.html)). Happy to provide input

Great, we can gladly discuss! I have some experience (or quite some, meanwhile, having it implemented in zfit) with pydantic too, and while I love aspects of it, there is one major drawback: it was never meant to be a serialization/deserialization or marshallization library in the first place; it's an input validation library. Unfortunately. See also [the discussion here](https://github.com/pydantic/pydantic/discussions/5389). Not sure with V2, I would love to use pydantic, but I think it's a different scope.
Nevertheless, we can both coordinate and work on this

**Comment by cburgard** [comment]
I'm very happy to see @nsmith- @jonas-eschle  starting the discussion on this important topic!
May I invite you to voice your opinion also here? #2 
The main point of argumentation there is the (ease of) implementing schema validation, so your input will likely be a make-or-break for that discussion!
